readthedocs/projects/forms.py

```diff
@@ -476,3 +464,29 @@ def save(self, **_):
             to_url=self.cleaned_data['to_url'],
         )
         return redirect
+
+
+class DomainForm(forms.ModelForm):
+
+    class Meta:
+        model = Domain
+        exclude = ['project', 'machine', 'cname', 'count']
+
+    def __init__(self, *args, **kwargs):
+        self.project = kwargs.pop('project', None)
+        super(DomainForm, self).__init__(*args, **kwargs)
+
+    def clean_canonical(self):
+        canonical = self.cleaned_data['canonical']
+        if canonical and Domain.objects.filter(
+            project=self.project, canonical=True
+        ).exclude(url=self.cleaned_data['url']).exists():
+            raise forms.ValidationError(_(u'Only 1 Domain can be canonical at a time.'))
+        return canonical
+
+    def save(self, *args, **kwargs):
```

Comment:
We're explicitly setting commit to False here, since it will fail without the project. I could just ignore what's passed in here.